### PR TITLE
Always serialize request params for GET request in dpApi

### PR DESCRIPTION
**Pull request by spiess-demos**
When moving from axios to fetch, it was overlooked that axios serializes params by default, and we only do that occasionally. This leads to the params being dropped in the fetch request whenever "serialize: true" is set via the options arg.

To overcome this, and as we always need to serialize params within get requests, the request params are always appended as stringified value to the url. As the "url" argument may already have params appended, this needs to be handled as well.

The "payload" variable has been renamed into "fetchOptions" to level up readability while distinguishing it from the "options" variable still being used in dpApi.

Also, "options" and "params" has been removed from the request payload as the fetch method does not use these values.

See https://developer.mozilla.org/en-US/docs/Web/API/fetch

### CODE REVIEW

File: src/lib/DpApi.js [hunk 1]
```diff
@@ -39,28 +39,34 @@ const getHeaders = function ({ headers, url }) {
...
+}
+
 const doRequest = (async ({ url, method = 'GET', data = {}, params, options = {} }) => {
-  const payload = {
-    method,
-    options,
-    params,
-    headers: getHeaders({ ...params, url })
+  const fetchOptions = {
+    headers: getHeaders({ ...params, url }),
```
Comment (salisdemos) [inline src/lib/DpApi.js:54]: Its confusing.
Shouldn't this be something like
```suggestion
    headers: getHeaders({ headers, url }),
```
and we have to  pass `headers` as another param to the method?
I know it was like that before. I just wonder how it ever worked. :see_no_evil:

File: src/lib/DpApi.js [hunk 2]
```diff
@@ -39,28 +39,34 @@ const getHeaders = function ({ headers, url }) {
...
+    method
   }
 
   if (method.toUpperCase() !== 'GET') {
     if (data instanceof FormData) {
-      payload.body = data
+      fetchOptions.body = data
```
Comment (salisdemos) [inline src/lib/DpApi.js:60]: why did You rename payload? at least, I think `fetchOptions` don't clarify what it is. Especially because `doRequest` does more than fetch ;-)

**Review by salisdemos — CHANGES_REQUESTED** [review]

File: src/lib/DpApi.js (hunk 1 above)
Comment (spiess-demos) [reply src/lib/DpApi.js:54]: it is a little bit confusing but what actually happens here is that all properties of `headers` are being spread into the object that is passed to `getHeaders`. otherwise you'd have to do something like

```
const { headers } = params
// or: const headers = params?.headers
headers: getHeaders({ headers, url })
```

and within `getHeaders` the object is destructured again, defining only the needed variables:

```
const getHeaders = function ({ headers, url }) {
```

Comment (salisdemos) [reply src/lib/DpApi.js:54]: But If You have a look where we pass Headers, they are not within params, but next to them ;-) 
like so:
```
      return dpApi({
        method: 'PATCH',
        url: 'someUrl',
        headers: {
          'Content-type': 'application/vnd.api+json',
          Accept: 'application/vnd.api+json'
        }
```

File: src/lib/DpApi.js (hunk 2 above)
Comment (hwiem) [reply src/lib/DpApi.js:60]: I agree that `fetchOptions` is slightly confusing (because it might also be a function name), but `payload` is also not a very descriptive name. We could change it to `settings` or `config`?

Comment (spiess-demos) [reply src/lib/DpApi.js:60]: The variable is passed as the second argument (the "options" argument) into the [`fetch()` global function](https://developer.mozilla.org/en-US/docs/Web/API/fetch#syntax). The decision to name that function "fetch", although it can send all kinds of requests, has been made by some other guys and i only intended to express "these are the options being passed into the fetch function" here.

**Review by salisdemos — APPROVED** [review]

**Review by hwiem — APPROVED** [review]

File: src/lib/DpApi.js (hunk 1 above)
Comment (spiess-demos) [reply src/lib/DpApi.js:54]: > I just wonder how it ever worked. 🙈

+1. i think that it did not, after dropping axios. It looks like the `headers` property in your last example (when using `dpApi` directly without the method aliases) was just ignored, as it is not forwarded to doRequest. as the api2defaultHeaders contain the correct headers anyways, this is just crucial for "a handful of 1.0 routes" that set custom headers.
